sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/v2Commands.scala

```diff
@@ -446,17 +501,52 @@ object DescribeColumn {
   def getOutputAttrs: Seq[Attribute] = DescribeCommandSchema.describeColumnAttributes()
 }
 
+trait RowLevelCommand extends Command with SupportsSubquery {
+  def condition: Expression
+  def rewritePlan: Option[LogicalPlan]
+  def withNewRewritePlan(newRewritePlan: LogicalPlan): RowLevelCommand
+}
+
 /**
  * The logical plan of the DELETE FROM command.
  */
 case class DeleteFromTable(
     table: LogicalPlan,
-    condition: Option[Expression]) extends UnaryCommand with SupportsSubquery {
-  override def child: LogicalPlan = table
-  override protected def withNewChildInternal(newChild: LogicalPlan): DeleteFromTable =
-    copy(table = newChild)
+    condition: Expression,
```

Comment:
@viirya @rdblue @cloud-fan @huaxingao, a question. In all rules I added, having checks for None and Some(true), which are essentially the same thing, complicated the implementation. I made the condition required and default it to true in the parser. What do you think?

Reply:
I think it's a good change.

Reply:
While I think it is better overall, I might not combine the DELETE FROM work with this change. It adds a lot of files to this diff.

Reply:
Yeah, I can make a separate PR for this if everyone is OK.

Reply:
Looks good to me too.